core/src/main/java/org/apache/calcite/sql/dialect/FireboltSqlDialect.java

```diff
@@ -196,4 +197,18 @@ public FireboltSqlDialect(Context context) {
       }
     }
   }
+
+  @Override public void unparseSqlIntervalLiteral(SqlWriter writer,
+      SqlIntervalLiteral literal, int leftPrec, int rightPrec) {
+    SqlIntervalLiteral.IntervalValue interval =
+        literal.getValueAs(SqlIntervalLiteral.IntervalValue.class);
+    writer.keyword("INTERVAL");
+    writer.print("'");
+    if (interval.getSign() == -1) {
+      writer.print("-");
+    }
+    writer.literal(interval.getIntervalLiteral());
+    writer.print(interval.getIntervalQualifier().toString());
```

Comment:
`toString` is AnsiSqlDialect.DEFAULT.  

I prefer to use `interval.getIntervalQualifier().unparse(writer, leftPrec, rightPrec);`

Reply:
I just realised that the implementation was a bit too simplistic and was not handling ranges (I just pushed a change)